### PR TITLE
Bump conjur authn k8s client

### PR DESCRIPTION
**Pull request by tzheleznyak**
Bump conjur authn k8s client to the version with JWT flow and refactor.

This PR fixes api changes of the new version of authn k8s client

### CODE REVIEW

File: cmd/secrets-provider/main.go [hunk 1]
```diff
@@ -120,29 +120,26 @@ func processAnnotations(ctx context.Context, tracer trace.Tracer) error {
 
 func secretRetriever(ctx context.Context,
 	tracer trace.Tracer) (*conjur.SecretRetriever, error) {
-
-	// Create a trace Span
-	spanCtx, span := tracer.Start(ctx, "Create Conjur secret retriever")
+	// Gather K8s authenticator config
```
Comment (sashaCher) [inline cmd/secrets-provider/main.go:123]: Is the intention there authn-k8s or configuration gathering is general for both authn's?

Comment (tzheleznyak) [reply cmd/secrets-provider/main.go:123]: General i will fix the comment

File: cmd/secrets-provider/main.go [hunk 2]
```diff
@@ -120,29 +120,26 @@ func processAnnotations(ctx context.Context, tracer trace.Tracer) error {
...
+	_, span := tracer.Start(ctx, "Gather authenticator config")
 	defer span.End()
 
-	// Gather K8s authenticator config
-	authnConfig, err := setupAuthnConfig(spanCtx, tracer)
+	authnConfig, err := authnConfigProvider.NewConfigFromCustomEnv(ioutil.ReadFile, customEnv)
 	if err != nil {
-		log.Error(err.Error())
 		span.RecordErrorAndSetStatus(err)
+		log.Error(messages.CSPFK008E)
```
Comment (Avi-Eyal) [inline cmd/secrets-provider/main.go:130]: is "CSPFK008E" the right error you want to use?

File: cmd/secrets-provider/main.go [hunk 3]
```diff
@@ -120,29 +120,26 @@ func processAnnotations(ctx context.Context, tracer trace.Tracer) error {
...
-		log.Error(err.Error())
+	if err = validateContainerMode(authnConfig.GetContainerMode()); err != nil {
 		span.RecordErrorAndSetStatus(err)
+		log.Error(err.Error())
 		return nil, err
 	}
 
 	// Initialize a Conjur secret retriever
-	secretRetriever, err := conjur.NewSecretRetriever(*authnConfig)
+	secretRetriever, err := conjur.NewSecretRetriever(authnConfig)
```
Comment (Avi-Eyal) [inline cmd/secrets-provider/main.go:140]: whats the difference ?

**Review by Avi-Eyal — DISMISSED** [review]

File: cmd/secrets-provider/main.go (hunk 2 above)
Comment (tzheleznyak) [reply cmd/secrets-provider/main.go:130]: CSPFK008E Failed to instantiate authenticator configuration

File: go.sum [hunk 4]
```diff
@@ -9,6 +9,10 @@ github.com/cyberark/conjur-api-go v0.8.0/go.mod h1:HZ5RoBhAB2KwnxyXbQ29DwpviRVg7
 github.com/cyberark/conjur-authn-k8s-client v0.22.1-0.20211129182011-dd5ac8ad32ff/go.mod h1:AUF2WFqMDwQevSqg+UM8qvbTxJ1Ij5RhUdDsNNYqB+c=
```
Comment (szh) [inline go.sum:9]: Please run `go mod tidy` to remove references to the old version

**Review by szh — CHANGES_REQUESTED** [review]
Looks good. Just left a minor comment

**Review by szh — APPROVED** [review]
Looks good, once you respond to the other change requests
👍

**Pull request merged**